### PR TITLE
Gas graph: add mouse interaction/dynamic proportions - HEL-110

### PR DESCRIPTION
**Pull request by nikolas**
Just the first steps here.

### CODE REVIEW

File: gas-retention-simulator/src/main.jsx [hunk 1]
```diff
@@ -114,37 +120,71 @@ class GasRetentionSimulator extends React.Component {
...
-            g.endFill();
-
-            if (this.state.activeGases.length === 1) {
-                g.position.x += 50;
-            } else if (this.state.activeGases.length === 2) {
-                g.position.x += 20;
-            }
 
-            app.stage.addChild(g);
+            this.drawGasBar(app, gas, i * 40);
```
Comment (nbuonin) [inline gas-retention-simulator/src/main.jsx]: `drawGasBars` and `drawGasBar` don't bind `this` in the constructor, is that ok?

Comment (nikolas) [reply gas-retention-simulator/src/main.jsx]: That's just how you refer to methods defined off the class here.. you could define them in the constructor, or ES6-style like I've done here.

```
class GasRetentionSimulator extends React.Component {
    drawGasBar() {
        // etc
    }
    drawGasBars() {
        // etc
    }
}
```

Comment (nbuonin) [reply gas-retention-simulator/src/main.jsx]: So is the classes `this` in scope for class methods that don't explicitly bind `this`? I guess I'm trying to understand for myself when I need to bind `this` and when I don't.

Comment (nbuonin) [reply gas-retention-simulator/src/main.jsx]: So yeah, looking at this example it looks like you don't: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Prototype_methods

Comment (nikolas) [reply gas-retention-simulator/src/main.jsx]: If you're talking about this kind of thing I'm doing in the constructor:
```
         this.onAddGas = this.onAddGas.bind(this);
         this.onRemoveGas = this.onRemoveGas.bind(this);
         this.onGasClick = this.onGasClick.bind(this);

         this.onGasBarClick = this.onGasBarClick.bind(this);
         this.onGasBarDragStart = this.onGasBarDragStart.bind(this);
         this.onGasBarDragEnd = this.onGasBarDragEnd.bind(this);
         this.onGasBarMove = this.onGasBarMove.bind(this);
```

That's necessary when binding event callbacks that need to refer to `this` in their function body. `drawGasBar()` is just a normal class method, so `this` inside this method will always refer to the instance of that class.

I don't fully understand why it's necessary to bind to this for event callbacks, that's just kind of a rule of thumb that I follow.

Comment (nbuonin) [reply gas-retention-simulator/src/main.jsx]: Yeah, thats my understanding too. So when you call `this.drawGasBar` here, the enclosing function doesn't bind `this`.

**Pull request merged**